src/dispatch/dispatcher.rs

```diff
@@ -2,13 +2,18 @@ use smallvec::SmallVec;
 
 use crate::{dispatch::stage::Stage, system::RunNow, world::World};
 
+/// This wrapper is used to share a replaceable ThreadPool with other
+/// dispatchers. Useful with batch dispatchers.
+#[derive(Debug, Default)]
+pub struct ThreadPoolWrapper(pub Option<::std::sync::Arc<::rayon::ThreadPool>>);
```

Comment:
Does this need to be newtype? Because the field is public it could be just type alias.

Reply:
Check my previous comment.

Reply:
I am not sure how the previous comment answers this. Amethyst or anyone else could still create `Option<::std::sync::Arc<::rayon::ThreadPool>>`. I don't see what value this new type adds as it doesn't restrict it's api (public field) and we're not implementing any traits for it (to get around orphan rules).
So instead of newtype it could just be:
```rust
type ThreadPoolWrapper = Option<::std::sync::Arc<::rayon::ThreadPool>>;
```

Reply:
The comment got sorted in the wrong way, so let me quote it here for clarity

> The outer `Arc` is necessary because the wrapper is shared across the dispatcher and its internal batches.
> Doing so allow change the `ThreadPool` to the main dispatcher and all its batches.
> 
> The internal `Arc` is needed because the `ThreadPool` could come from someone that is not using the Wrapper, for example the Amethyst thread pool.


Basically using a wrapper as I did we have this situation:

```
Dispatcher ━━━━━━━━━━━━━━━━┓
   ├ Batch ━━━━━━━━━━━━━━ Wrapper ━━━━━━━>ThreadPool 1
   ├ Batch ━━━━━━━━━━━━━━━┛┃
   └ Batch ━━━━━━━━━━━━━━━━┛
```
When I set a new `ThreadPool` in the `Dispatcher` I change the inner part of the wrapper and the result will be this:

```
Dispatcher ━━━━━━━━━━━━━━━━┓
   ├ Batch ━━━━━━━━━━━━━━ Wrapper ━━━━━━━>ThreadPool 2
   ├ Batch ━━━━━━━━━━━━━━━┛┃
   └ Batch ━━━━━━━━━━━━━━━━┛
```

------------------
If instead I use the type like you advice (which is what actually shred use) I'll be in this state:
```
Dispatcher ━━━━━━━━━━━━━━━━┓
   ├ Batch ━━━━━━━━━━━━━━ ThreadPool 1
   ├ Batch ━━━━━━━━━━━━━━━┛┃
   └ Batch ━━━━━━━━━━━━━━━━┛
```
If I set a new `ThreadPool` in the `Dispatcher` only the Dispatcher will have it:
```
Dispatcher ━━━━━━━━━━━━━━ ThreadPool 2
   ├ Batch ━━━━━━━━━━━━━━ ThreadPool 1
   ├ Batch ━━━━━━━━━━━━━━━┛┃
   └ Batch ━━━━━━━━━━━━━━━━┛
```

Reply:
Am I doing something wrong here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c43559e03501eb998c1dd2e4ba82ee69

Reply:
You forgot to use `ThreadWrapperPool` check this https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8d0df726d3e72398a355453d95cc2ba4

Reply:
And this does the same: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=349fcdcb498585cf0a4a4d8209b3c00b

Reply:
Correct, let me change it!

Reply:
Done!